### PR TITLE
Refactor vgpr occupancy calculation for Aldebaran's new hardware capability Unified Register Files 

### PR DESCRIPTION
**Pull request by jichangjichang**
For UnifiedRegisterFiles, arch vgpr and acc vgpr share same pool of 512 registers.
This PR refactor the getOccupany calculation of KernelWriter and extend getOccupancy CI tests.

### CODE REVIEW

File: Tensile/KernelWriterAssembly.py [hunk 1]
```diff
@@ -861,6 +841,34 @@ def selectMemoryInstruction(self,
     else:
       raise RuntimeError("Could not find valid memory instruction for operation=%s, width=%f, kernel=%s" %(operation, width, self.kernelName))
 
+  class VgprOccupancyCurve:
```
Comment (rosenrodt) [inline Tensile/KernelWriterAssembly.py:844]: Any particular motivation to move the class into `KernelWriterAssembly`?

**Review by rosenrodt — CHANGES_REQUESTED** [review]
Please check the failing tests in Tensile/Tests/unit/test_KernelWriterAssembly.py

Some of the relevant functions are made either a static method of `KernelWriterAssembly` or independent from `KernelWriterAssembly` is because of ease of test. So you may want to revisit the your changes that moved those function into `KernelWriterAssembly`.

**Review by jichangjichang — COMMENTED** [review]
> Please check the failing tests in Tensile/Tests/unit/test_KernelWriterAssembly.py
> 
> Some of the relevant functions are made either a static method of `KernelWriterAssembly` or independent from `KernelWriterAssembly` is because of ease of test. So you may want to revisit the your changes that moved those function into `KernelWriterAssembly`.

This test should be removed.
Occupancy table is different between different GCN version.
Test should not treat it as a static table.

**Comment by rosenrodt** [comment]
> This test should be removed.
> Occupancy table is different between different GCN version.
> Test should not treat it as a static table.

The updated occupancy calculator still needs to be proven correct one way or the other. Removing the tests won't help with that goal.

You can use `pytest.mark.parametrize` decorator for arch-dependent tests. There is also a predefined pytest fixture `useGlobalParameters` in conftest.py to allow you to access `globalParameters` from within the test. That should be enough

**Comment by jichangjichang** [comment]
> > This test should be removed.
> > Occupancy table is different between different GCN version.
> > Test should not treat it as a static table.
> 
> The updated occupancy calculator still needs to be proven correct one way or the other. Removing the tests won't help with that goal.
> 
> You can use `pytest.mark.parametrize` decorator for arch-dependent tests. There is also a predefined pytest fixture `useGlobalParameters` in conftest.py to allow you to access `globalParameters` from within the test. That should be enough

Designed has been changed. Occupancy is calculated when runtime but not refer to a table.
The CI test scope also extend for UnifiedRegisterFiles.

**Comment by jichangjichang** [comment]
> > This test should be removed.
> > Occupancy table is different between different GCN version.
> > Test should not treat it as a static table.
> 
> The updated occupancy calculator still needs to be proven correct one way or the other. Removing the tests won't help with that goal.
> 
> You can use `pytest.mark.parametrize` decorator for arch-dependent tests. There is also a predefined pytest fixture `useGlobalParameters` in conftest.py to allow you to access `globalParameters` from within the test. That should be enough

**Comment by jichangjichang** [comment]
> Please check the failing tests in Tensile/Tests/unit/test_KernelWriterAssembly.py
> 
> Some of the relevant functions are made either a static method of `KernelWriterAssembly` or independent from `KernelWriterAssembly` is because of ease of test. So you may want to revisit the your changes that moved those function into `KernelWriterAssembly`.

@rosenrodt need your review to unblock the change requested

**Review by rosenrodt — APPROVED** [review]
Nice. Looks good

**Comment by jichangjichang** [comment]
rocblas-test pass on gfx908 and gfx90a

**Review by zaliu — APPROVED** [review]

**Pull request merged**